plugin/audit_log/buffer.c

```diff
@@ -163,11 +167,25 @@ void audit_log_buffer_shutdown(audit_log_buffer_t *log)
 int audit_log_buffer_write(audit_log_buffer_t *log, const char *buf, size_t len)
 {
   if (len > log->size)
-    return(1);
+  {
+    if (!log->drop_if_full)
+    {
+      mysql_mutex_lock(&log->mutex);
```

Comment:
What user would need is means to ascertain whether the problem needs addressing or is one-of. I'd probably add a status counter and diagnose the first occurrence to the error log. Obviously not in this PR